src/main/flight/mixer.c

```diff
@@ -613,11 +619,30 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
 
         }
 #endif
+
+        // reduce motorRangeMax when battery is full
+        if (vbatSagCompensationFactor > 0.0f) {
+            const float vbatFull = batteryConfig()->vbatmaxcellvoltage - 10;
```

Comment:
The default max cell voltage for a normal LiPo is 4.3V.  That's about 0.1V higher than the typical real full battery voltage, 4.2V, which is where the upper bound of the compensation range should start.  

HV cells are nominally charged to 4.35V and I think that a suitable max cell voltage for reliable cell detection is usually set to 0.1V above this by most users.  

In both cases, subtracting 0.1V off the max cell voltage gives us a suitable value for the upper bound of the compensation range.

Reply:
What are you doing for users that have `vbatmaxcellvoltage` set to 4.5V because they occasionally fly HV LiPo, but are currently flying a 'normal' LiPo?
I think simplifying this by removing the `- 10` will reduce the complexity for no disadvantage.

Reply:
This is a very good question, with no easy answer.

The max value sets the range, which in turn sets the peak percentage motor output reduction. It doesn't so strongly affect what happens at lower battery voltages.
Empirically, with normal LiPo batteries, the code seems to work very well when the maximum range value is 4.2V, ie 0.1V under the default battery max value.  

If a whoop user, for example, was to set an HV max at 4.45V, that would also work optimally, with the max being set, appropriately, at 4.35V.

We should encourage users to set the max for HV at 0.1V above the nominal value, ie to 4.45V.  This code would then work optimally when flown with HV batteries. 

Let's say a user chops and changes between batteries, leaving the battery max at 4.45V all the time.  When they really have an HV battery, the compensation will make it feel like a normal Lipo (ie a normal lipo with a normal default max voltage value).  However when they use a stock LiPo, they won't get any more compensation unless they drop the max voltage, so it will feel a bit less aggressive.

In the normal situation, that's exactly what would happen, anyway, without this code, since shifting to an HV always adds responsiveness over a stock LiPo, and vice versa.  So the outcome is no different.

Clearly, this code can only compensate properly if it knows which kind of battery is connected.  The only way they can have absolutely consistent 'feel', regardless of battery type, is to manually change their max cell value down to 4.3V for their normal LiPO flights, - and bring it up again to 4.45V for their HV flights.  

For pilots who mostly fly standard LiPo cells but occasionally fly HV cells, they can leave the max at 4.45V, and set their compensation factor to 82.  This will result in normal (as per default) feel when they use 4.2V lipo's, and the quad will be a bit more aggressive when HV cells are used, which is the exact normal situation for people who do that.